src/types/Event.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  DurationEventHandler,
+  DurationEventsPhase,
+  DurationEventInterface,
+} from './DurationEventHandler';
+
+import {
+  CompleteEventHandler,
+  CompleteEventsPhase,
+  CompleteEventInterface,
+} from './CompleteEventsHandler';
+
+export interface EventInterface {
+  name: string;
+  cat: string;
+  ts: number;
+  pid: number;
+  tid: number;
+  ph: string;
+  sf?: number;
+  cname?: string;
+}
+
+export abstract class EventHandler {
+  constructor(public row: EventInterface) {}
+  abstract process(): EventInterface;
+}
+
+export class Event {
+  handler!: EventHandler;
```

Comment:
@saphal1998 why do we need the `!` here? Is it to tell TypeScript this field is non-null? And if so, is that necessary because the field is only assigned a value in the constructor?

Reply:
So, basically I am delegating all the work to the handler, and I create seperate handlers based on the type of event it is (which I do by checking the phase).
The reason I use the ```!``` is simply because VSCode was giving errors without it. And I think it makes sense, because we have to **absolutely must assign** a handler to an event. I'm sure there is a better way to enforce this.